httpclient5/src/main/java/org/apache/hc/client5/http/impl/auth/DigestScheme.java

```diff
@@ -253,6 +254,16 @@ public String generateAuthResponse(
         if (this.paramMap.get("nonce") == null) {
             throw new AuthenticationException("missing nonce");
         }
+
+        if (context != null) {
+            final HttpClientContext clientContext = HttpClientContext.cast(context);
+            final String nextNonce = clientContext.getAttribute("auth-nextnonce", String.class);
```

Comment:
@arturobernalg Maybe this deserves being an attribute in the context. Up to you.

Reply:
@ok2c  that is a good idea. I made all the necessary change.